### ISSUE TITLE
Rework loading of Chatterino badges

### ISSUE THREAD

**Issue by Felanbird** [Chatterino/chatterino2#3118]
**Describe your issue**

Chatterino badges are only loaded on startup, it might be nice to look into reworking this, as the most common issue seen with it is with new Supporters who don't see their own badge, while users who recently opened Chatterino, do.

**Screenshots**
![image](https://user-images.githubusercontent.com/41973452/128254607-eaa17001-89b5-4ce5-b5a1-ae6872a8f3be.png)

![image](https://user-images.githubusercontent.com/41973452/128254654-c979acc1-56b4-475e-9f0c-f2cd4f2c18ce.png)

Screenshots from discussion involving new addition to Developer badge, logic applies to Supporter & Contributor as well. 

**OS and Chatterino Version**
N/A

**Comment by LosFarmosCTL** [comment]
Could take a look at it, but how exactly should this be implemented? The badges are retrieved from a normal REST api right? So should there be some kind of timer to fetch an updated badge list every n seconds? 
Or should it be integrated into some other reloading mechanism already present (seems like the better option to me, but the question still remains which one, maybe the emote reloading since that already reloads FFZ mod/vip badges?)

**Comment by xHeaveny** [comment]
> Could take a look at it, but how exactly should this be implemented? The badges are retrieved from a normal REST api right? So should there be some kind of timer to fetch an updated badge list every n seconds?
> Or should it be integrated into some other reloading mechanism already present (seems like the better option to me, but the question still remains which one, maybe the emote reloading since that already reloads FFZ mod/vip badges?)

I guess with reloading the FFZ and BTTV Emotes this can also be reloaded so

**Comment by zneix** [comment]
> I guess with reloading the FFZ and BTTV Emotes this can also be reloaded so

I don't agree with this. Reloading with F5 already takes into account many items - all twitch emotes fetched from both kraken and helix. Channel emotes from all the services and if we keep adding things, like Global third party emotes / ffz / bttv (/ 7tv) badges it's gonna get very messy in my opinion.

Another point against the issue in general, Chatterino badges change rather very rarely, especially developer/contributor ones. So I'd rather consider these as something constant. As for supporter badges, you're already explicitly told on the [wiki](https://wiki.chatterino.com/Help/#how-do-i-get-the-supporter-badge) to restart your Chatterino in order to see the badge. When someone donates it's probably also rather a one-time action, so if there's no way to manually reload these badges it shouldn't be a big deal.
If anything, there could be a separate option to reload only Chatterino badges (or badges in general, if we were to add a way to reload other badges, e.g. ffz ones). Why? This is not something you wanna reload/refresh so often as the emotes and I think calling Chatterino API every time you wanna refresh only emotes is unnecessary.

**Comment by leon-richardt** [comment]
> If anything, there could be a separate option to reload only Chatterino badges (or badges in general, if we were to add a way to reload other badges, e.g. ffz ones).

I strongly disagree with this. Chatterino should strive for a simpler UI instead of complicating things by introducing many different shortcuts for very similar things (from the user's perspective, at least).

Also, if the API request frequency is of concern, we could always ratelimit client-side.

**Comment by LosFarmosCTL** [comment]
> Chatterino should strive for a simpler UI instead of complicating things by introducing many different shortcuts for very similar things (from the user's perspective, at least).

Especially for something like this I don't think that an extra button makes any sense. If a user is already dedicated enough to manually reload supporter badges via some sub-menu button, he might as well just restart chatterino. 
The problem, at least from my perspective, is not so much how difficult it is to reload the badges, but that it is a dedicated step someone has to consciously take.
